File: storage/data_original/at_coder/abc079/C/4891430.py
s = input()
A,B,C,D = s[0],s[1],s[2],s[3]

for op1,op2,op3 in [("+","+","+"),("+","+","-"),("+","-","+"),("+","-","-"),("-","+","+"),("-","+","-"),("-","-","+"),("-","-","-")]:
    if eval(A+op1+B+op2+C+op3+D) == 7:
        print(A+op1+B+op2+C+op3+D+"=7")
        break

File: storage/data_transpiled/at_coder/abc079/C/4891430.java
public static void input ( Scanner s ) {
  String A = s . next ( ) ;
  String B = s . next ( ) ;
  String C = s . next ( ) ;
  String D = s . next ( ) ;
  for ( int i = 0 ;
  i < 4 ;
  i ++ ) {
    String op1 = s . next ( ) ;
    String op2 = s . next ( ) ;
    String op3 = s . next ( ) ;
    if ( new Integer ( A + op1 + B + op2 + C + op3 + D ) . intValue ( ) == 7 ) {
      System . out . println ( A + op1 + B + op2 + C + op3 + D + "=7" ) ;
      break ;
    }
  }
}
